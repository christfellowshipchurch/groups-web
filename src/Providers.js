import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
