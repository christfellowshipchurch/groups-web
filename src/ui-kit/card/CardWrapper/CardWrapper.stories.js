import React from 'react';
import { storiesOf } from '@storybook/react';

import CardWrapper from '.';

storiesOf('ui-kit/Card/CardWrapper', module)
  .addDecorator((story) => <div style={{ width: '50%' }}>{story()}</div>)
  .add('default', () => <CardWrapper>Boom!</CardWrapper>)
  .add('cardColor', () => {
    const styles = {
      paddingTop: '56.25%',
      width: '100%',
    };
    return <CardWrapper cardColor={'salmon'} style={styles} />;
  });
