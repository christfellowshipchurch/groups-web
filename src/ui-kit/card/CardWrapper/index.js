import styled from 'styled-components';
import PropTypes from 'prop-types';

// import { withIsLoading } from '../../isLoading';

const CardWrapper = styled.div.attrs({
  className: 'ui-kit-card-CardWrapper',
})`
  /* card styles */
  background: ${({ cardColor, theme }) =>
    cardColor ||
    theme.colors.background.paper ||
    `initial`}; /* bail out if no bg color */
  border-radius: ${({ theme }) => theme.sizing.baseBorderRadius};
  margin: ${({ inHorizontalList, theme }) =>
    inHorizontalList
      ? `${theme.sizing.baseUnit(0.5)}
      ${theme.sizing.baseUnit(0.5)}
      ${theme.sizing.baseUnit(0.75)}`
      : `${theme.sizing.baseUnit(0.75)} ${theme.sizing.baseUnit(1)}`};
  box-shadow: ${({ theme }) => theme.shadows.default};
`;

CardWrapper.propTypes = {
  cardColor: PropTypes.string,
  children: PropTypes.node.isRequired,
};

CardWrapper.displayName = 'ui-kit.card.CardWrapper';

export default CardWrapper;
