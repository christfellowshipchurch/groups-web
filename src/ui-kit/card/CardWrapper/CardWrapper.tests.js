import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import CardWrapper from '.';

describe('the CardWrapper component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <CardWrapper>Boom!</CardWrapper>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept a backgroundColor', () => {
    const tree = renderer.create(
      <Providers>
        <CardWrapper cardColor={'salmon'}>Boom!</CardWrapper>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const cardDimensions = {
      paddingTop: '56.25%',
      width: '100%',
    };

    const tree = renderer.create(
      <Providers>
        <CardWrapper style={cardDimensions}>Boom!</CardWrapper>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <CardWrapper role={'article listitem'}>Boom!</CardWrapper>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
