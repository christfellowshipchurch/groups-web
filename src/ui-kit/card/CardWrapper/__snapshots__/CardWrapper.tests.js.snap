// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CardWrapper component should accept a backgroundColor 1`] = `
.c0 {
  background: salmon;
  border-radius: 16px;
  margin: 0.75rem 1rem;
  box-shadow: 0px 5px 18px rgba(0,0,0,0.1);
}

<div
  className="c0 ui-kit.card.CardWrapper"
>
  Boom!
</div>
`;

exports[`the CardWrapper component should accept additional props 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 16px;
  margin: 0.75rem 1rem;
  box-shadow: 0px 5px 18px rgba(0,0,0,0.1);
}

<div
  className="c0 ui-kit.card.CardWrapper"
  role="article listitem"
>
  Boom!
</div>
`;

exports[`the CardWrapper component should accept and render passed in styles 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 16px;
  margin: 0.75rem 1rem;
  box-shadow: 0px 5px 18px rgba(0,0,0,0.1);
}

<div
  className="c0 ui-kit.card.CardWrapper"
  style={
    Object {
      "paddingTop": "56.25%",
      "width": "100%",
    }
  }
>
  Boom!
</div>
`;

exports[`the CardWrapper component should render 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 16px;
  margin: 0.75rem 1rem;
  box-shadow: 0px 5px 18px rgba(0,0,0,0.1);
}

<div
  className="c0 ui-kit.card.CardWrapper"
>
  Boom!
</div>
`;
